### PR TITLE
Storybook integration

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Storybook integration - This mini-project provides an overview of the different components of the interface of the "Upgrade assistant" module in different versions of Prestashop.
| Type?             | new feature
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | Follow the steps in the readme to set up storybook. The components should appear as in the attached video. The autoupgrade module itself should work the same as before

## BC breaks

- *.twig files have been renamed to *.html.twig, to be consistent with CORE

## To-do
- [x] Rename twig files
- [x] Make sure module still works in PS context
- [x] Implement most complex template to try out `trans` modifier, inclusion of other template and aliases
- [x] Include theme(s) of PS 8+
- [x] Make sure README is complete
- [x] Remove images from themes
- [x] Add prettier
- [x] Remove storybook folder on release build

## To-do (later)
* Add translations management
* Move form generation from PHP class in templates
* Mock ajax requests

[Capture vidéo du 08-07-2024 14:24:12.webm](https://github.com/PrestaShop/autoupgrade/assets/36233446/aef28e50-cc8e-4a6b-8439-5861b4ab8e75)

### CODE REVIEW

File: storybook/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Storybook overview
+
+This mini-project provides an overview of the different components of the interface of the "Upgrade assistant" module
```
Comment (ga-devfront) [inline storybook/README.md:3]:
```suggestion
This mini-project provides an overview of the different components of the interface of the "Update assistant" module
```

**Comment by Quetzacoalt91** [comment]
Waiting for a decision to take about the themes included in the autoupgrade project or not.

**Comment by Quetzacoalt91** [comment]
Unblocked: This PR will have the themes assets removed from the commits list to avoid pushing to much development stuff in the history, then another PR will be created later to add them via a NPM dependency.

**Comment by Quetzacoalt91** [comment]
>BC breaks?

Yes, templates extensions have changed

File: storybook/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+- Node.js >= 19 - [Download Node.js](https://nodejs.org/)
+
+## Install project dependencies
+
+**Install PHP dependencies**
+
+To install the necessary PHP dependencies, run the following command:
+
+```shell
+$ composer install
```
Comment (Quetzacoalt91) [inline storybook/README.md:19]:
```suggestion
# From the storybook/ folder:
$ composer install
```

File: views/templates/macros/icons.html.twig [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{% macro ok(psBaseUri) %}
+    <img src="{{ psBaseUri }}img/admin/enabled.gif" alt="ok">
+{%  endmacro %}
+{% macro nok(psBaseUri) %}
+    <img src="{{ psBaseUri }}img/admin/disabled.gif" alt="nok">
```
Comment (Quetzacoalt91) [inline views/templates/macros/icons.html.twig:5]: What if the variable was set in the parent template? Do we still have to send it as parameter?

File: views/templates/block/checklist.html.twig [hunk 4]
```diff
@@ -202,7 +202,7 @@
                         {% endfor %}
                       </ul>
                     </td>
-                    <td>{{ icons.nok }}</td>
+                    <td>{{ icons.nok(psBaseUri) }}</td>
```
Comment (Quetzacoalt91) [inline views/templates/block/checklist.html.twig:205]: I don't really like that we have to send the variable `psBaseUri` to each call, is there any other way?

File: views/templates/macros/icons.html.twig (hunk 3 above)
Comment (M0rgan01) [reply views/templates/macros/icons.html.twig:5]: https://twig.symfony.com/doc/3.x/tags/macro.html

> But as with PHP functions, macros don't have access to the current template variables.

We must pass by a parameter

File: views/templates/block/checklist.html.twig (hunk 4 above)
Comment (M0rgan01) [reply views/templates/block/checklist.html.twig:205]: I'm not very comfortable with this approach too :/ another solution is to use the _context variable, (_context.psBaseUri) but this is worse for me

**Review by Quetzacoalt91 — DISMISSED** [review]

File: views/templates/block/checklist.html.twig (hunk 4 above)
Comment (Quetzacoalt91) [reply views/templates/block/checklist.html.twig:205]: Too bad. Marking as solved.

File: views/templates/macros/icons.html.twig (hunk 3 above)
Comment (Quetzacoalt91) [reply views/templates/macros/icons.html.twig:5]: Too bad it is not applicable. Marking as solved.

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as we expected 

Tested from : 
1.7.8.8 to 8.1.6
8.0.5 to 8.1.6
8.1.5 to 8.1.6
8.1.6 to 9.0.0
8.1.5 to 9.0.0
8.0.5 to 9.0.0 


Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**